satpy/tests/reader_tests/test_eum_l2_grib.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019 Satpy developers
+#
+# satpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# satpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""EUM L2 GRIB-reader test package."""
+
+import datetime
+import sys
+import unittest
+from unittest import mock
+
+import numpy as np
+
+from satpy.tests.utils import make_dataid
+
+# Dictionary to be used as fake GRIB message
+FAKE_SEVIRI_MESSAGE = {
+    "longitudeOfSubSatellitePointInDegrees": 9.5,
+    "dataDate": 20191020,
+    "dataTime": 1745,
+    "Nx": 1000,
+    "Ny": 1200,
+    "earthMajorAxis": 6400.,
+    "earthMinorAxis": 6300.,
+    "NrInRadiusOfEarth": 6.,
+    "XpInGridLengths": 500,
+    "parameterNumber": 30,
+    "missingValue": 9999,
+}
+
+FAKE_FCI_MESSAGE = {
+    "longitudeOfSubSatellitePointInDegrees": 0.0,
+    "dataDate": 20191020,
+    "dataTime": 1745,
+    "Nx": 5568,
+    "Ny": 5568,
+    "earthMajorAxis": 6378140.,
+    "earthMinorAxis": 6356755.,
+    "NrInRadiusOfEarth": 6.6107,
+    "XpInGridLengths": 2784.0,
+    "parameterNumber": 30,
+    "missingValue": 9999,
+}
+
+# List to be used as fake GID source
+FAKE_GID = [0, 1, 2, 3, None]
+
+
+class Test_EUML2GribFileHandler(unittest.TestCase):
```

Comment:
It's quite a lot of duplicate code for the SEVIRI and FCI tests, can you please have a look and see if this can be refactored and optimized a bit?

Reply:
I moved the duplicated code to a common method, please check if it is necessary to do something else